ncm-network/src/main/perl/networkmanager.pm

```diff
@@ -0,0 +1,724 @@
+#${PMpre} NCM::Component::networkmanager${PMpost}
+
+=head1 NAME
+
+network: Extention of Network to configure Network settings using NetworkManager by configuring in Keyfile format.
+Most functions and logic is taken from network module to minimise changes to current network module.
+
+=head1 DESCRIPTION
+
+The I<network> component sets the network settings through C<< /etc/sysconfig/network >>
+and the NM keyfile settings in files C<< /etc/NetworkManager/system-connections >>.
+
+New/changed settings are first tested by retrieving the latest profile from the
+CDB server (using ccm-fetch). 
+If this fails, the component reverts all settings to the previous values. This is no different to network module.
+
+During this test, a sleep value of 15 seconds is used to make sure the restarted network
+is fully restarted (routing may need some time to come up completely).
+
+Because of this, configuration changes may cause the ncm-ncd run to take longer than usual.
+
+Be aware that configuration changes can also lead to a brief network interruption.
+=cut
+
+use parent qw (NCM::Component::network);
+
+our $EC = LC::Exception::Context->new->will_store_all;
+
+our $NoActionSupported = 1;
+
+use Readonly;
+
+Readonly my $HOSTNAME_CMD => '/usr/bin/hostnamectl';
+
+Readonly my $NETWORK_PATH => '/system/network';
+Readonly my $HARDWARE_PATH => '/hardware/cards/nic';
+Readonly my $NMCFG_DIR => "/etc/NetworkManager/system-connections";
+Readonly my $BACKUP_DIR => "$NMCFG_DIR/.quattorbackup";
+Readonly my $NETWORKCFG => "/etc/sysconfig/network";
+
+Readonly my $RESOLV_CONF => '/etc/resolv.conf';
+Readonly my $RESOLV_CONF_SAVE => '/etc/resolv.conf.save';
+Readonly my $RESOLV_SUFFIX => '.ncm-network';
+
+# need this to remain same value as what is in network.pm
```

Comment:
you can import the values or perhaps use them straight away. there should be no need for this.